src/main/java/org/jboss/tm/listener/AssociatingEventType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.tm.listener;
+
+/**
+ * An indication of an event that effects the current transaction
+ */
+public enum AssociatingEventType implements LifecycleEventType<AssociatingEventType> {
+    /**
+     * a transaction is about to be associated with the current thread
+     */
+    ASSOCIATING,
```

Comment:
It should have been in a test package. I added it in order to test that I could add new event types and I used it in the test code and in the BaseTransactionManagerDelegate code but the latter needs removing. I will come up with a test case that does not require modifying  BaseTransactionManagerDelegate